# Heart Disease Health Indicators Dataset Notebook

## Purpose
The purpose of this code notebook is to clean BRFSS data into a useable format for machine learning alogrithms. 
The dataset originally has 330 features (columns), but based on heart disease research regarding factors influencing heart disease and other chronic health conditions, only select features are included in this analysis.

## Link to Dataset Output [Heart Disease Health Indicators Dataset](https://www.kaggle.com/alexteboul/heart-disease-health-indicators-dataset)
**253,680 survey responses from cleaned BRFSS 2015 - binary classification**

#### Important Risk Factors
Research in the field has identified the following as **important risk factors** for heart disease and other chronic illnesses like diabetes (not in strict order of importance):

*   blood pressure (high)
*   cholesterol (high)
*   smoking
*   diabetes
*   obesity
*   age
*   sex
*   race
*   diet
*   exercise
*   alcohol consumption
*   BMI
*   Household Income
*   Marital Status
*   Sleep
*   Time since last checkup
*   Education
*   Health care coverage
*   Mental Health

### Selected Subset of Features from BRFSS 2015
Given these risk factors, I tried to select features (columns/questions) in the BRFSS related to these risk factors. To help understand what the columns mean, I consult the BRFSS 2015 Codebook to see the questions and information about the questions. I try to match the variable names in the codebook to the variable names in the dataset I downloaded from Kaggle. I also reference some of the same features chosen for a research paper by Zidian Xie et al for *Building Risk Prediction Models for Type 2 Diabetes Using Machine Learning Techniques* using the 2014 BRFSS. Diabetes and Heart Disease outcomes are strongly correlated, with the primary cause of death for diabetics being heart disease complications. Given this information, it is a useful starting point.

**BRFSS 2015 Codebook:** https://www.cdc.gov/brfss/annual_data/2015/pdf/codebook15_llcp.pdf

**Relevant Research Paper using BRFSS for Diabetes ML:** https://www.cdc.gov/pcd/issues/2019/19_0109.htm


The **selected features** from the BRFSS 2015 dataset are:

**Response Variable / Dependent Variable:**
*   Respondents that have ever reported having coronary heart disease (CHD) or myocardial infarction (MI) --> _MICHD


**Independent Variables:**

**High Blood Pressure**
*   Adults who have been told they have high blood pressure by a doctor, nurse, or other health professional --> _RFHYPE5

**High Cholesterol**
*   Have you EVER been told by a doctor, nurse or other health professional that your blood cholesterol is high? --> TOLDHI2
*   Cholesterol check within past five years --> _CHOLCHK

**BMI**
*   Body Mass Index (BMI) --> _BMI5

**Smoking**
*   Have you smoked at least 100 cigarettes in your entire life? [Note: 5 packs = 100 cigarettes] --> SMOKE100

**Other Chronic Health Conditions**
*   (Ever told) you had a stroke. --> CVDSTRK3
*   (Ever told) you have diabetes (If "Yes" and respondent is female, ask "Was this only when you were pregnant?". If Respondent says pre-diabetes or borderline diabetes, use response code 4.) --> DIABETE3

**Physical Activity**
*   Adults who reported doing physical activity or exercise during the past 30 days other than their regular job --> _TOTINDA

**Diet**
*   Consume Fruit 1 or more times per day --> _FRTLT1
*   Consume Vegetables 1 or more times per day --> _VEGLT1

**Alcohol Consumption**
*   Heavy drinkers (adult men having more than 14 drinks per week and adult women having more than 7 drinks per week) --> _RFDRHV5

**Health Care**
*   Do you have any kind of health care coverage, including health insurance, prepaid plans such as HMOs, or government plans such as Medicare, or Indian Health Service?  --> HLTHPLN1
*   Was there a time in the past 12 months when you needed to see a doctor but could not because of cost? --> MEDCOST

**Health General and Mental Health**
*   Would you say that in general your health is: --> GENHLTH
*   Now thinking about your mental health, which includes stress, depression, and problems with emotions, for how many days during the past 30 days was your mental health not good? --> MENTHLTH
*   Now thinking about your physical health, which includes physical illness and injury, for how many days during the past 30 days was your physical health not good? --> PHYSHLTH
*   Do you have serious difficulty walking or climbing stairs? --> DIFFWALK

**Demographics**
*   Indicate sex of respondent. --> SEX
*   Fourteen-level age category --> _AGEG5YR
*   What is the highest grade or year of school you completed? --> EDUCA
*   Is your annual household income from all sources: (If respondent refuses at any income level, code "Refused.") --> INCOME2

## 1. Get the data

In [102]:
#imports
import os
import pandas as pd
import random
random.seed(1)

In [103]:
brfss_2015_dataset = pd.read_csv("C:/Users/duvallar/OneDrive/1.RoseHulman/3.Junior/Winter/MA384/Data/indicators_data_dirty/2015.csv")

In [104]:
#How many rows and columns
brfss_2015_dataset.shape

(441456, 330)

In [105]:
#check that the data loaded in is in the correct format
pd.set_option('display.max_columns', 500)
brfss_2015_dataset.head()

_STATE  FMONTH        IDATE IMONTH   IDAY    IYEAR  DISPCODE         SEQNO  \
0     1.0     1.0  b'01292015'  b'01'  b'29'  b'2015'    1200.0  2.015000e+09   
1     1.0     1.0  b'01202015'  b'01'  b'20'  b'2015'    1100.0  2.015000e+09   
2     1.0     1.0  b'02012015'  b'02'  b'01'  b'2015'    1200.0  2.015000e+09   
3     1.0     1.0  b'01142015'  b'01'  b'14'  b'2015'    1100.0  2.015000e+09   
4     1.0     1.0  b'01142015'  b'01'  b'14'  b'2015'    1100.0  2.015000e+09   

           _PSU  CTELENUM  PVTRESD1  COLGHOUS  STATERES  CELLFON3  LADULT  \
0  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   
1  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   
2  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   
3  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   
4  2.015000e+09       1.0       1.0       NaN       1.0       2.0     NaN   

   NUMADULT        NUMMEN  NUMWOMEN  CTELNUM1  CELLFON2  CADULT  PVTRESD2  \
0       3.0  1.000000e+00       2.0       NaN       NaN     NaN       NaN   
1       1.0  5.397605e-79       1.0       NaN       NaN     NaN       NaN   
2       2.0  1.000000e+00       1.0       NaN       NaN     NaN       NaN   
3       3.0  1.000000e+00       2.0       NaN       NaN     NaN       NaN   
4       2.0  1.000000e+00       1.0       NaN       NaN     NaN       NaN   

   CCLGHOUS  CSTATE  LANDLINE  HHADULT  GENHLTH  PHYSHLTH  MENTHLTH  POORHLTH  \
0       NaN     NaN       NaN      NaN      5.0      15.0      18.0      10.0   
1       NaN     NaN       NaN      NaN      3.0      88.0      88.0       NaN   
2       NaN     NaN       NaN      NaN      4.0      15.0      88.0      88.0   
3       NaN     NaN       NaN      NaN      5.0      30.0      30.0      30.0   
4       NaN     NaN       NaN      NaN      5.0      20.0      88.0      30.0   

   HLTHPLN1  PERSDOC2  MEDCOST  CHECKUP1  BPHIGH4  BPMEDS  BLOODCHO  CHOLCHK  \
0       1.0       1.0      2.0       1.0      1.0     1.0       1.0      1.0   
1       2.0       1.0      1.0       4.0      3.0     NaN       1.0      4.0   
2       1.0       2.0      2.0       1.0      3.0     NaN       1.0      1.0   
3       1.0       2.0      1.0       1.0      1.0     1.0       1.0      1.0   
4       1.0       1.0      2.0       1.0      3.0     NaN       1.0      1.0   

   TOLDHI2  CVDINFR4  CVDCRHD4  CVDSTRK3  ASTHMA3  ASTHNOW  CHCSCNCR  \
0      1.0       2.0       2.0       2.0      1.0      1.0       2.0   
1      2.0       2.0       2.0       2.0      2.0      NaN       2.0   
2      1.0       7.0       2.0       1.0      2.0      NaN       2.0   
3      1.0       2.0       2.0       2.0      2.0      NaN       2.0   
4      2.0       2.0       2.0       2.0      2.0      NaN       2.0   

   CHCOCNCR  CHCCOPD1  HAVARTH3  ADDEPEV2  CHCKIDNY  DIABETE3  DIABAGE2  SEX  \
0       2.0       1.0       1.0       1.0       2.0       3.0       NaN  2.0   
1       2.0       2.0       2.0       2.0       2.0       3.0       NaN  2.0   
2       1.0       2.0       1.0       2.0       2.0       3.0       NaN  2.0   
3       1.0       2.0       1.0       1.0       2.0       3.0       NaN  2.0   
4       2.0       2.0       1.0       2.0       2.0       3.0       NaN  2.0   

   MARITAL  EDUCA  RENTHOM1  NUMHHOL2  NUMPHON2  CPDEMO1  VETERAN3  EMPLOY1  \
0      1.0    4.0       1.0       2.0       NaN      1.0       2.0      8.0   
1      2.0    6.0       1.0       2.0       NaN      2.0       2.0      3.0   
2      2.0    4.0       1.0       2.0       NaN      1.0       2.0      7.0   
3      1.0    4.0       1.0       2.0       NaN      1.0       2.0      8.0   
4      1.0    5.0       1.0       2.0       NaN      2.0       2.0      8.0   

   CHILDREN  INCOME2  INTERNET  WEIGHT2  HEIGHT3  PREGNANT  QLACTLM2  \
0      88.0      3.0       2.0    280.0    510.0       NaN       1.0   
1      88.0      1.0       1.0    165.0    508.0       NaN       1.0   
2      88.0     9

**At this point we have 441,456 records and 330 columns. Each record contains an individual's BRFSS survey responses.**

In [106]:
# select specific columns
brfss_df_selected = brfss_2015_dataset[['_MICHD', 
                                         '_RFHYPE5',  
                                         'TOLDHI2', '_CHOLCHK', 
                                         '_BMI5', 
                                         'SMOKE100', 
                                         'CVDSTRK3', 'DIABETE3', 
                                         '_TOTINDA', 
                                         '_FRTLT1', '_VEGLT1', 
                                         '_RFDRHV5', 
                                         'HLTHPLN1', 'MEDCOST', 
                                         'GENHLTH', 'MENTHLTH', 'PHYSHLTH', 'DIFFWALK', 
                                         'SEX', '_AGEG5YR', 'EDUCA', 'INCOME2' ]]

In [107]:
brfss_df_selected.shape

(441456, 22)

In [108]:
brfss_df_selected.head()

_MICHD  _RFHYPE5  TOLDHI2  _CHOLCHK   _BMI5  SMOKE100  CVDSTRK3  DIABETE3  \
0     2.0       2.0      1.0       1.0  4018.0       1.0       2.0       3.0   
1     2.0       1.0      2.0       2.0  2509.0       1.0       2.0       3.0   
2     NaN       1.0      1.0       1.0  2204.0       NaN       1.0       3.0   
3     2.0       2.0      1.0       1.0  2819.0       2.0       2.0       3.0   
4     2.0       1.0      2.0       1.0  2437.0       2.0       2.0       3.0   

   _TOTINDA  _FRTLT1  _VEGLT1  _RFDRHV5  HLTHPLN1  MEDCOST  GENHLTH  MENTHLTH  \
0       2.0      2.0      1.0       1.0       1.0      2.0      5.0      18.0   
1       1.0      2.0      2.0       1.0       2.0      1.0      3.0      88.0   
2       9.0      9.0      9.0       9.0       1.0      2.0      4.0      88.0   
3       2.0      1.0      2.0       1.0       1.0      1.0      5.0      30.0   
4       2.0      9.0      1.0       1.0       1.0      2.0      5.0      88.0   

   PHYSHLTH  DIFFWALK  SEX  _AGEG5YR  EDUCA  INCOME2  
0      15.0       1.0  2.0       9.0    4.0      3.0  
1      88.0       2.0  2.0       7.0    6.0      1.0  
2      15.0       NaN  2.0      11.0    4.0     99.0  
3      30.0       1.0  2.0       9.0    4.0      8.0  
4      20.0       2.0  2.0       9.0    5.0     77.0

## 2. Clean the data

### 2.1 Drop missing values

In [109]:
#Drop Missing Values - knocks 100,000 rows out right away
original_row_count = brfss_df_selected.shape[0]
brfss_df_selected = brfss_df_selected.dropna()
droped_row_count = brfss_df_selected.shape[0]
print(brfss_df_selected.shape, "Percentage Droped:  ", ((original_row_count -  droped_row_count)/original_row_count) * 100)


(343606, 22) Percentage Droped:   22.165289405965712


### 2.2 Modify and clean the values to be more suitable to ML algorithms
In order to do this part, I referenced the codebook which says what each column/feature/question is: https://www.cdc.gov/brfss/annual_data/2015/pdf/codebook15_llcp.pdf

In [110]:
# _MICHD
#Change 2 to 0 because this means did not have MI or CHD
brfss_df_selected['_MICHD'] = brfss_df_selected['_MICHD'].replace({2: 0})
print(brfss_df_selected._MICHD.unique())


[0. 1.]


In [111]:
#1 _RFHYPE5
#Change 1 to 0 so it represetnts No high blood pressure and 2 to 1 so it represents high blood pressure
brfss_df_selected['_RFHYPE5'] = brfss_df_selected['_RFHYPE5'].replace({1:0, 2:1})
brfss_df_selected = brfss_df_selected[brfss_df_selected._RFHYPE5 != 9]
print(brfss_df_selected._RFHYPE5.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[1. 0.]
(342900, 22)     Dropped percent:  0.21     Droped rows:    706


In [112]:
#2 TOLDHI2
# Change 2 to 0 because it is No
# Remove all 7 (dont knows)
# Remove all 9 (refused)
brfss_df_selected['TOLDHI2'] = brfss_df_selected['TOLDHI2'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected.TOLDHI2 != 7]
brfss_df_selected = brfss_df_selected[brfss_df_selected.TOLDHI2 != 9]
print(brfss_df_selected.TOLDHI2.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[1. 0.]
(340178, 22)     Dropped percent:  1.0     Droped rows:    3428


In [113]:
#3 _CHOLCHK
# Change 3 to 0 and 2 to 0 for Not checked cholesterol in past 5 years
# Remove 9
brfss_df_selected['_CHOLCHK'] = brfss_df_selected['_CHOLCHK'].replace({3:0,2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected._CHOLCHK != 9]
print(brfss_df_selected._CHOLCHK.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[1. 0.]
(335825, 22)     Dropped percent:  2.26     Droped rows:    7781


In [114]:
#4 _BMI5 (no changes, just note that these are BMI * 100. So for example a BMI of 4018 is really 40.18)
brfss_df_selected['_BMI5'] = brfss_df_selected['_BMI5'].div(100).round(0)
brfss_df_selected._BMI5.unique()

array([40., 25., 28., 24., 27., 30., 26., 23., 34., 33., 21., 22., 31.,
       38., 20., 19., 32., 46., 41., 37., 36., 29., 35., 18., 54., 45.,
       39., 47., 43., 55., 49., 42., 17., 16., 48., 44., 50., 59., 15.,
       52., 53., 57., 51., 14., 58., 63., 61., 56., 60., 74., 62., 64.,
       13., 66., 73., 65., 68., 85., 71., 84., 67., 70., 82., 79., 92.,
       72., 88., 96., 81., 12., 77., 95., 75., 91., 69., 76., 87., 89.,
       83., 98., 86., 80., 90., 78., 97.])

In [115]:
#5 SMOKE100
# Change 2 to 0 because it is No
# Remove all 7 (dont knows)
# Remove all 9 (refused)
brfss_df_selected['SMOKE100'] = brfss_df_selected['SMOKE100'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected.SMOKE100 != 7]
brfss_df_selected = brfss_df_selected[brfss_df_selected.SMOKE100 != 9]
print(brfss_df_selected.SMOKE100.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[1. 0.]
(333858, 22)     Dropped percent:  2.84     Droped rows:    9748


In [116]:
#6 CVDSTRK3
# Change 2 to 0 because it is No
# Remove all 7 (dont knows)
# Remove all 9 (refused)
brfss_df_selected['CVDSTRK3'] = brfss_df_selected['CVDSTRK3'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected.CVDSTRK3 != 7]
brfss_df_selected = brfss_df_selected[brfss_df_selected.CVDSTRK3 != 9]
print(brfss_df_selected.CVDSTRK3.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[0. 1.]
(333120, 22)     Dropped percent:  3.05     Droped rows:    10486


In [117]:
#7 DIABETE3
# going to make this ordinal. 0 is for no diabetes or only during pregnancy, 1 is for pre-diabetes or borderline diabetes, 2 is for yes diabetes
# Remove all 7 (dont knows)
# Remove all 9 (refused)
brfss_df_selected['DIABETE3'] = brfss_df_selected['DIABETE3'].replace({2:0, 3:0, 1:2, 4:1})
brfss_df_selected = brfss_df_selected[brfss_df_selected.DIABETE3 != 7]
brfss_df_selected = brfss_df_selected[brfss_df_selected.DIABETE3 != 9]
print(brfss_df_selected.DIABETE3.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[0. 2. 1.]
(332801, 22)     Dropped percent:  3.14     Droped rows:    10805


In [118]:
#8 _TOTINDA
# 1 for physical activity
# change 2 to 0 for no physical activity
# Remove all 9 (don't know/refused)
brfss_df_selected['_TOTINDA'] = brfss_df_selected['_TOTINDA'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected._TOTINDA != 9]
print(brfss_df_selected._TOTINDA.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[0. 1.]
(317404, 22)     Dropped percent:  7.63     Droped rows:    26202


In [119]:
#9 _FRTLT1
# Change 2 to 0. this means no fruit consumed per day. 1 will mean consumed 1 or more pieces of fruit per day 
# remove all dont knows and missing 9
brfss_df_selected['_FRTLT1'] = brfss_df_selected['_FRTLT1'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected._FRTLT1 != 9]
print(brfss_df_selected._FRTLT1.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[0. 1.]
(309768, 22)     Dropped percent:  9.85     Droped rows:    33838


In [120]:
#10 _VEGLT1
# Change 2 to 0. this means no vegetables consumed per day. 1 will mean consumed 1 or more pieces of vegetable per day 
# remove all dont knows and missing 9
brfss_df_selected['_VEGLT1'] = brfss_df_selected['_VEGLT1'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected._VEGLT1 != 9]
print(brfss_df_selected._VEGLT1.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[1. 0.]
(302160, 22)     Dropped percent:  12.06     Droped rows:    41446


In [121]:
#11 _RFDRHV5
# Change 1 to 0 (1 was no for heavy drinking). change all 2 to 1 (2 was yes for heavy drinking)
# remove all dont knows and missing 9
brfss_df_selected['_RFDRHV5'] = brfss_df_selected['_RFDRHV5'].replace({1:0, 2:1})
brfss_df_selected = brfss_df_selected[brfss_df_selected._RFDRHV5 != 9]
print(brfss_df_selected._RFDRHV5.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[0. 1.]
(298637, 22)     Dropped percent:  13.09     Droped rows:    44969


In [122]:
#12 HLTHPLN1
# 1 is yes, change 2 to 0 because it is No health care access
# remove 7 and 9 for don't know or refused
brfss_df_selected['HLTHPLN1'] = brfss_df_selected['HLTHPLN1'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected.HLTHPLN1 != 7]
brfss_df_selected = brfss_df_selected[brfss_df_selected.HLTHPLN1 != 9]
print(brfss_df_selected.HLTHPLN1.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[1. 0.]
(298145, 22)     Dropped percent:  13.23     Droped rows:    45461


In [123]:
#13 MEDCOST
# Change 2 to 0 for no, 1 is already yes
# remove 7 for don/t know and 9 for refused
brfss_df_selected['MEDCOST'] = brfss_df_selected['MEDCOST'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected.MEDCOST != 7]
brfss_df_selected = brfss_df_selected[brfss_df_selected.MEDCOST != 9]
print(brfss_df_selected.MEDCOST.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[0. 1.]
(297709, 22)     Dropped percent:  13.36     Droped rows:    45897


In [124]:
#14 GENHLTH
# This is an ordinal variable that I want to keep (1 is Excellent -> 5 is Poor)
# Remove 7 and 9 for don't know and refused
brfss_df_selected = brfss_df_selected[brfss_df_selected.GENHLTH != 7]
brfss_df_selected = brfss_df_selected[brfss_df_selected.GENHLTH != 9]
print(brfss_df_selected.GENHLTH.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[5. 3. 2. 4. 1.]
(297194, 22)     Dropped percent:  13.51     Droped rows:    46412


In [125]:
#15 MENTHLTH
# already in days so keep that, scale will be 0-30
# change 88 to 0 because it means none (no bad mental health days)
# remove 77 and 99 for don't know not sure and refused
brfss_df_selected['MENTHLTH'] = brfss_df_selected['MENTHLTH'].replace({88:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected.MENTHLTH != 77]
brfss_df_selected = brfss_df_selected[brfss_df_selected.MENTHLTH != 99]
print(brfss_df_selected.MENTHLTH.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[18.  0. 30.  3.  5. 15. 10.  6. 20.  2. 25.  1. 29.  4.  7.  8. 21. 14.
 26.  9. 16. 28. 11. 12. 24. 17. 13. 23. 27. 19. 22.]
(294102, 22)     Dropped percent:  14.41     Droped rows:    49504


In [126]:
#16 PHYSHLTH
# already in days so keep that, scale will be 0-30
# change 88 to 0 because it means none (no bad mental health days)
# remove 77 and 99 for don't know not sure and refused
brfss_df_selected['PHYSHLTH'] = brfss_df_selected['PHYSHLTH'].replace({88:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected.PHYSHLTH != 77]
brfss_df_selected = brfss_df_selected[brfss_df_selected.PHYSHLTH != 99]
print(brfss_df_selected.PHYSHLTH.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[15.  0. 30.  2. 14. 28.  7. 20.  3. 10.  1.  5. 17.  4. 19.  6. 21. 12.
  8. 25. 27. 22. 29. 24.  9. 16. 18. 23. 13. 26. 11.]
(290473, 22)     Dropped percent:  15.46     Droped rows:    53133


In [127]:
#17 DIFFWALK
# change 2 to 0 for no. 1 is already yes
# remove 7 and 9 for don't know not sure and refused
brfss_df_selected['DIFFWALK'] = brfss_df_selected['DIFFWALK'].replace({2:0})
brfss_df_selected = brfss_df_selected[brfss_df_selected.DIFFWALK != 7]
brfss_df_selected = brfss_df_selected[brfss_df_selected.DIFFWALK != 9]
print(brfss_df_selected.DIFFWALK.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[1. 0.]
(289698, 22)     Dropped percent:  15.69     Droped rows:    53908


In [128]:
#18 SEX
# in other words - is respondent male (somewhat arbitrarily chose this change because men are at higher risk for heart disease)
# change 2 to 0 (female as 0). Male is 1
brfss_df_selected['SEX'] = brfss_df_selected['SEX'].replace({2:0})
brfss_df_selected.SEX.unique()

array([0., 1.])

In [129]:
#19 _AGEG5YR
# already ordinal. 1 is 18-24 all the way up to 13 wis 80 and older. 5 year increments.
# remove 14 because it is don't know or missing
brfss_df_selected = brfss_df_selected[brfss_df_selected._AGEG5YR != 14]
print(brfss_df_selected._AGEG5YR.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[ 9.  7. 11. 10. 13.  8.  4.  6.  2. 12.  5.  1.  3.]
(288259, 22)     Dropped percent:  16.11     Droped rows:    55347


In [130]:
#20 EDUCA
# This is already an ordinal variable with 1 being never attended school or kindergarten only up to 6 being college 4 years or more
# Scale here is 1-6
# Remove 9 for refused:
brfss_df_selected = brfss_df_selected[brfss_df_selected.EDUCA != 9]
print(brfss_df_selected.EDUCA.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[4. 6. 3. 5. 2. 1.]
(287931, 22)     Dropped percent:  16.2     Droped rows:    55675


In [131]:
#21 INCOME2
# Variable is already ordinal with 1 being less than $10,000 all the way up to 8 being $75,000 or more
# Remove 77 and 99 for don't know and refused
brfss_df_selected = brfss_df_selected[brfss_df_selected.INCOME2 != 77]
brfss_df_selected = brfss_df_selected[brfss_df_selected.INCOME2 != 99]
print(brfss_df_selected.INCOME2.unique())
print(brfss_df_selected.shape, "    Dropped percent: " , percentDropped(brfss_df_selected), "    Droped rows:   ", droped_row_count - brfss_df_selected.shape[0])


[3. 1. 8. 6. 4. 7. 2. 5.]
(253680, 22)     Dropped percent:  26.17     Droped rows:    89926


In [132]:
#Check the shape of the dataset now: We have 253,680 cleaned rows and 22 columns (1 of which is our dependent variable)
print(brfss_df_selected.shape,  "Percent Dropped From Start: ", round(((original_row_count - brfss_df_selected.shape[0]) / original_row_count) * 100, 2), " Original Row Count: ", original_row_count)

(253680, 22) Percent Dropped From Start:  42.54  Original Row Count:  441456


In [133]:
#Let's see what the data looks like after Modifying Values
brfss_df_selected.head()

_MICHD  _RFHYPE5  TOLDHI2  _CHOLCHK  _BMI5  SMOKE100  CVDSTRK3  DIABETE3  \
0     0.0       1.0      1.0       1.0   40.0       1.0       0.0       0.0   
1     0.0       0.0      0.0       0.0   25.0       1.0       0.0       0.0   
3     0.0       1.0      1.0       1.0   28.0       0.0       0.0       0.0   
5     0.0       1.0      0.0       1.0   27.0       0.0       0.0       0.0   
6     0.0       1.0      1.0       1.0   24.0       0.0       0.0       0.0   

   _TOTINDA  _FRTLT1  _VEGLT1  _RFDRHV5  HLTHPLN1  MEDCOST  GENHLTH  MENTHLTH  \
0       0.0      0.0      1.0       0.0       1.0      0.0      5.0      18.0   
1       1.0      0.0      0.0       0.0       0.0      1.0      3.0       0.0   
3       0.0      1.0      0.0       0.0       1.0      1.0      5.0      30.0   
5       1.0      1.0      1.0       0.0       1.0      0.0      2.0       0.0   
6       1.0      1.0      1.0       0.0       1.0      0.0      2.0       3.0   

   PHYSHLTH  DIFFWALK  SEX  _AGEG5YR  EDUCA  INCOME2  
0      15.0       1.0  0.0       9.0    4.0      3.0  
1       0.0       0.0  0.0       7.0    6.0      1.0  
3      30.0       1.0  0.0       9.0    4.0      8.0  
5       0.0       0.0  0.0      11.0    3.0      6.0  
6       0.0       0.0  0.0      11.0    5.0      4.0

In [134]:
#Check Class Sizes of the heart disease column  Note the class imbalance!
print(brfss_df_selected.groupby(['_MICHD']).size())
grp =brfss_df_selected.groupby(['_MICHD'])
percent_yes = 100*(grp.size()[1.0] / grp.size().sum())
print("Percent Yes (1.0): ",  round(percent_yes, 2))

_MICHD
0.0    229787
1.0     23893
dtype: int64
Percent Yes (1.0):  9.42


## 3. Make feature names more readable

In [135]:
#Rename the columns to make them more readable
brfss = brfss_df_selected.rename(columns = {'_MICHD':'HeartDiseaseorAttack', 
                                         '_RFHYPE5':'HighBP',  
                                         'TOLDHI2':'HighChol', '_CHOLCHK':'CholCheck', 
                                         '_BMI5':'BMI', 
                                         'SMOKE100':'Smoker', 
                                         'CVDSTRK3':'Stroke', 'DIABETE3':'Diabetes', 
                                         '_TOTINDA':'PhysActivity', 
                                         '_FRTLT1':'Fruits', '_VEGLT1':"Veggies", 
                                         '_RFDRHV5':'HvyAlcoholConsump', 
                                         'HLTHPLN1':'AnyHealthcare', 'MEDCOST':'NoDocbcCost', 
                                         'GENHLTH':'GenHlth', 'MENTHLTH':'MentHlth', 'PHYSHLTH':'PhysHlth', 'DIFFWALK':'DiffWalk', 
                                         'SEX':'Sex', '_AGEG5YR':'Age', 'EDUCA':'Education', 'INCOME2':'Income' })

In [136]:
brfss.head()

HeartDiseaseorAttack  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0                   0.0     1.0       1.0        1.0  40.0     1.0     0.0   
1                   0.0     0.0       0.0        0.0  25.0     1.0     0.0   
3                   0.0     1.0       1.0        1.0  28.0     0.0     0.0   
5                   0.0     1.0       0.0        1.0  27.0     0.0     0.0   
6                   0.0     1.0       1.0        1.0  24.0     0.0     0.0   

   Diabetes  PhysActivity  Fruits  Veggies  HvyAlcoholConsump  AnyHealthcare  \
0       0.0           0.0     0.0      1.0                0.0            1.0   
1       0.0           1.0     0.0      0.0                0.0            0.0   
3       0.0           0.0     1.0      0.0                0.0            1.0   
5       0.0           1.0     1.0      1.0                0.0            1.0   
6       0.0           1.0     1.0      1.0                0.0            1.0   

   NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  Education  \
0          0.0      5.0      18.0      15.0       1.0  0.0   9.0        4.0   
1          1.0      3.0       0.0       0.0       0.0  0.0   7.0        6.0   
3          1.0      5.0      30.0      30.0       1.0  0.0   9.0        4.0   
5          0.0      2.0       0.0       0.0       0.0  0.0  11.0        3.0   
6          0.0      2.0       3.0       0.0       0.0  0.0  11.0        5.0   

   Income  
0     3.0  
1     1.0  
3     8.0  
5     6.0  
6     4.0

In [137]:
brfss.shape

(253680, 22)

In [138]:
#Check how many respondents have had heart disease or a heart attack. Note the class imbalance!
brfss.groupby(['HeartDiseaseorAttack']).size()

HeartDiseaseorAttack
0.0    229787
1.0     23893
dtype: int64

## 4. Save to csv
First save version where heart disease is the target variable and in the first column, then save one where diabetes is the target variable and in the first column.

In [139]:
#************************************************************************************************
brfss.to_csv("C:/Users/duvallar/OneDrive/1.RoseHulman/3.Junior/Winter/MA384/Data/indicators_data_clean/indicators_BRFSS2015.csv" , sep=",", index=False)
#************************************************************************************************